src/PaymentGateways/Gateways/Stripe/Actions/CreateCheckoutSession.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Give\PaymentGateways\Gateways\Stripe\Actions;
+
+use Give\PaymentGateways\DataTransferObjects\GatewayPaymentData;
+use Give\PaymentGateways\Gateways\Stripe\ValueObjects\CheckoutSession;
+use Give\PaymentGateways\Gateways\Stripe\ValueObjects\DonationSummary;
+use Give\PaymentGateways\Gateways\Stripe\ValueObjects\PaymentMethod;
+use Give\PaymentGateways\Gateways\Stripe\WorkflowAction;
+use Give\ValueObjects\Money;
+
+/**
+ * @unreleased
+ */
+class CreateCheckoutSession extends WorkflowAction
+{
+    /**
+     * @unreleased
+     * @param GatewayPaymentData $paymentData
+     * @param DonationSummary $donationSummary
+     * @param \Give_Stripe_Customer $giveStripeCustomer
+     * @param PaymentMethod $paymentMethod
+     * @return void
+     */
+    public function __invoke(
+        GatewayPaymentData $paymentData,
+        DonationSummary $donationSummary,
+        \Give_Stripe_Customer $giveStripeCustomer,
+        PaymentMethod $paymentMethod
+    )
+    {
+
+        $session_args = [
+            'customer'                   => $giveStripeCustomer->get_id(),
+            'client_reference_id'        => $paymentData->purchaseKey,
+            'payment_method_types'       => [ 'card' ],
+            'billing_address_collection' => give_is_setting_enabled( give_get_option( 'stripe_collect_billing' ) ) ? 'required' : 'auto',
+            'mode'                       => 'payment',
+            'line_items'                 => [
+                [
+                    'name'        => give_get_donation_form_title( $paymentData->donationId ),
+                    'description' => $donationSummary->getSummary(),
+                    'amount'      => Money::of( $paymentData->price, $paymentData->currency )->getMinorAmount(),
+                    'currency'    => $paymentData->currency,
+                    'quantity'    => 1,
+                ],
+            ],
+            'payment_intent_data'        => [
+                'capture_method'       => 'automatic',
+                'description'          => $donationSummary->getSummary(),
+                'metadata'             => give_stripe_prepare_metadata( $paymentData->donationId ),
+                'statement_descriptor' => give_stripe_get_statement_descriptor(),
+            ],
+            'submit_type'                => 'donate',
+            'success_url'                => give_get_success_page_uri(),
+            'cancel_url'                 => give_get_failed_transaction_uri(),
```

Comment:
@kjohnson Is there any reason we are not using the gateway route?

Reply:
@ravinderk the sessions args were copy/pasted from the legacy implementation - I hadn't considered refactoring the url properties to use gateway routes. That is a good idea, which I can address after this PR is merged.

Reply:
@kjohnson 👍